### DISCUSSION TITLE
Workspace icons for folders and places should be themable

### DISCUSSION THREAD

**Discussion by BertrandDekoninck** [trunkmaster/nextspace#392] [Ideas]
You choose not to use stock GNUstep icons for folders and places. As a consequence of this, they are not themable. See the attached screenshot of GWorkspace and Workspace side by side with the rik theme.
![gworkspace vs workspace](https://user-images.githubusercontent.com/12986802/33520300-7954af9a-d7b8-11e7-9eaa-2bf173132cc0.png)

I think your icons should be provided by a theme (Nextspace theme ?), and not hard-codded in Workspace. This way, Workspace could be coherent with any theme. This theme, I think I could contribute, by collecting the icons and putting them together in the theme.

**Comment by trunkmaster** [comment]
Hmm... I may be a solution for problem we disscussed in maillist. Let it be.

**Comment by BertrandDekoninck** [comment]
I've played a little with this  tonight. I've taken the Heritage theme, and put inside the icons I've taken in Workspace resources directory. You can see the differences between the original theme and the modified one  on GWorkspaces hereafter :
![gworkspace next](https://user-images.githubusercontent.com/12986802/33630839-07358f3a-da09-11e7-836a-26cb32c2fcd0.png)

![gworkspace heritage](https://user-images.githubusercontent.com/12986802/33630872-1a96e01a-da09-11e7-8ee9-181c42375b1e.png)
I've not yet found in Workspace's code how your images are associated to the different type of location. Is it in Workspace or in one NX framework ?
What you need first for Workspace and NXAppKit  is a CFBundleIdentifier in their  Info-gnustep.plist
You can for instance add this in it : 
` CFBundleIdentifier =  "org.nextspace.Workspace"; `

 (GNUstep uses the "org.gnustep" domain, gap uses "org.gap")
This way, in any theme, we can add a org.nextspace.Workspace folder containing all the "named images" of Workspace, meaning the images you made themable  in your code .
In the Nextspace.theme prototype attached, I've created  this folder in the Resources theme, containing several images that could be themed in the code.
[theme.zip](https://github.com/trunkmaster/nextspace/files/1533019/theme.zip)

Note that one version of these images should still be shipped in Wokspace itself, so that themes that don't have the org.nextspace.Workspace folder won't show broken icons.

**Comment by trunkmaster** [comment]
Thank you. I'll think about it later.

**Comment by BertrandDekoninck** [comment]
Bad news. I just tried to compiled GWorkspace in my NEXTSPACE vm: it fails because gui is too old :  gui doesn't know about NSImageNameMultipleDocuments, which is part of the API to theme named images.

For instance, Rik.theme builds fine but can't either be used by applications.

So before adapting Workspace to let its icons be themed, you should upgrade gnustep to 0.25 first.
.

**Comment by trunkmaster** [comment]
I'll update gnustep libraries later, after reaching set of features/stability I've planned for v1.0.
